syntax/ocaml.vim

```diff
@@ -86,11 +87,20 @@ syn region   ocamlNone transparent matchgroup=ocamlEncl start="{" matchgroup=oca
 syn region   ocamlNone transparent matchgroup=ocamlEncl start="\[" matchgroup=ocamlEncl end="\]" contains=ALLBUT,@ocamlContained,ocamlBrackErr
 syn region   ocamlNone transparent matchgroup=ocamlEncl start="\[|" matchgroup=ocamlEncl end="|\]" contains=ALLBUT,@ocamlContained,ocamlArrErr
 
-
-" Comments
-syn region   ocamlComment start="(\*" end="\*)" contains=@Spell,ocamlComment,ocamlTodo
+" Comments and documentation
 syn keyword  ocamlTodo contained TODO FIXME XXX NOTE
 
+syn cluster ocamlCommentLike contains=ocamlComment,ocamlCommentInDoc,ocamlDocumentation,ocamlEmptyDocumentation
+
+if !exists('odoc_syntax_loading')
+  " Load odoc syntax only when we are not in a *.mld file
+  syn region   ocamlComment start="(\*" end="\*)" contains=@Spell,ocamlComment,ocamlTodo
+  syn include  @ocamlOdoc syntax/odoc.vim
+  syn region   ocamlDocumentation matchgroup=ocamlComment start="(\*\*" end="\*)" contains=@ocamlOdoc
+  syn match    ocamlEmptyDocumentation "(\*\*)"
+else
+  syn region   ocamlCommentInDoc start="(\*" end="\*)" contains=@Spell,ocamlTodo,ocamlCommentInDoc
```

Comment:
We ended up duplicating it because it seemed that the vim grammar was getting confused with only the one `ocamlComment`. It might be worth rechecking that it is indeed still necessary.

Reply:
We tried this modification on the `.ml` file you provide above, it makes vim open two ocamlComments regions at the first `(*`, so we are stuck in comments for the whole file.

Tried with vim 9.1 and neovim 0.9.4